src/filter/wrap.rs

```diff
@@ -1,27 +1,21 @@
 use super::Filter;
 
-pub trait WrapSealed<F: Filter> {
+/// Wraps filters.
```

Comment:
can we give a detailed explanation of the trait, in the line of others like [reply](https://github.com/seanmonstar/warp/blob/master/src/reply.rs#L198) with a working example?

Reply:
I'm struggling to come up with an example use case. Most existing uses of the `Wrap` trait use some sealed structure, so I'm not sure what would make sense to demonstrate here.

Reply:
it's getting nice! can you show where you are stalled? I think you can use `BoxedFilter`

Reply:
I'm struggling to come up with a terse, but plausible use for `Wrap`. I look to something like `warp::filters::log::Log` or `warp::filters::reply::WithHeader` and see that their `Wrap` implementation is relying on private items. I don't think `BoxedFilter`s will make this task easier for items that are like `Log` or `WithHeader`

The ultimate motivation behind me wanting to unseal this trait is to be able to implement something like what is detailed in this comment: https://github.com/seanmonstar/warp/pull/408#issuecomment-606325196, but I see that that also relies on other private items.

So its a problem of 1. I don't know what the example should demonstrate. and 2. I don't know if a plausible example could be created with the other "primitives" that are exposed currently.